rust/lance-core/src/error.rs

```diff
@@ -87,6 +104,7 @@ impl From<ArrowError> for Error {
     fn from(e: ArrowError) -> Self {
         Self::Arrow {
             message: e.to_string(),
+            location: location!(),
```

Comment:
This location is unlikely to be very useful but I'm not sure we can do much better since `From` doesn't give us the option of forwarding the location.